Tests/SwiftProtobufTests/TestHelpers.swift

```diff
@@ -42,6 +42,22 @@ extension PBTestHelpers where MessageTestType: SwiftProtobuf.Message & Equatable
             } catch {
                 XCTFail("Failed to decode protobuf: \(string(from: encoded))", file: file, line: line)
             }
+
+            let expectedSize = try configured.serializedBinarySize()
```

Comment:
Please add tests that use `serializeBinary(into:)` with a few more cases:

- a buffer a little too short and a message with fields set to ensure that a proper error is thrown (looks like `BinaryEncodingError` doesn't have something for "too short", so the current code might not support this case (just assert), so it might take a lot of plumbing to get this right.
- a zero length buffer and a message with no fields set (since the binary size will be zero) and ensure it passes
- a zero length buffer and a message with fields set to ensure that does error.
- a buffer larger than needed and a message with fields set to ensure it works and reports the correct size.
- a buffer larger than needed and a message with no fields set to ensure it works and reports zero.